test/autostart.js

```diff
@@ -0,0 +1,32 @@
+var tape = require('tape');
+var queue = require('../');
+
+tape('autostart', function(t) {
+  t.plan(4);
+  var actual = [];
+  var q = queue({ autostart: true });
+
+  q.push(function(cb) {
+    actual.push('one');
+    cb();
+  });
+  q.push(function (cb) {
+    actual.push('two');
+    cb();
+  });
+  setTimeout(function () {
+    q.push(function (cb) {
+      actual.push('three');
+      cb();
+    });
+  }, 1000);
+
+  q.on('end', function() {
```

Comment:
I'm still a bit confused as to why the `end` event is only triggered once, even though the third task is done asynchronously.